crates/router/src/core/payments.rs

```diff
@@ -3391,6 +3504,37 @@ where
     Ok(merchant_connector_account)
 }
 
+#[cfg(feature = "v2")]
+#[instrument(skip_all)]
+pub async fn construct_profile_id_and_get_mca<'a, F, D>(
```

Comment:
we do not need this function, only mca id is required to get the `merchant_connector_account`.

Reply:
Also the return type should be MerchantConnectorAccount domain model, not from the helpers